src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -245,8 +262,14 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
 
             // Create ObjectTypeDefinitionNode for linking entities. These object definitions are not exposed in the schema
             // but are used to generate the object definitions of directional linking entities for (source, target) and (target, source) entities.
-            Dictionary<string, ObjectTypeDefinitionNode> linkingObjectTypes = GenerateObjectDefinitionsForLinkingEntities();
-            GenerateSourceTargetLinkingObjectDefinitions(objectTypes, linkingObjectTypes);
+            // However, ObjectTypeDefinitionNode for linking entities are need only for multiple create operation. So, creating these only when multiple create operations are
+            // enabled.
+            if (_isMultipleCreateOperationEnabled)
+            {
+
```

Comment:
nit: remove blank line

Reply:
I don't think this needs to be put under a conditional. The conditional logic is already there in SqlMetadataProvider to  create linking entities when multiple create is enabled: https://github.com/Azure/data-api-builder/blob/814bc34f8e6f3a79e194af62f4e71a1137e5482b/src/Core/Services/MetadataProviders/SqlMetadataProvider.cs#L761

Trying to put as less conditionals/branches as we can - will make life easier when we/other devs add support for other db types.

Reply:
ObjectTypeDefinitionNode objects are created for linking tables only when multiple create is enabled. Adding a conditional in `GraphQLSchemaCreator` makes it easier to follow rather than letting the flow go through and a condition failing at a later stage preventing the creation of  ObjectTypeDefinitionNode objects.